indexing-service/src/main/java/org/apache/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -388,12 +389,26 @@ private long jitter(long input)
     return new ArrayList<>(timeline.values());
   }
 
-  private void initializeSplitsIfNeeded()
+  private void initializeSplitsIfNeeded(@Nullable SplitHintSpec splitHintSpec)
   {
     if (splits != null) {
       return;
     }
 
+    final SegmentsSplitHintSpec nonNullSplitHintSpec;
+    if (!(splitHintSpec instanceof SegmentsSplitHintSpec)) {
+      if (splitHintSpec != null) {
+        log.warn("Given splitHintSpec[%s] is not a SegmentsSplitHintSpec. Ingoring it.", splitHintSpec);
+      }
+      nonNullSplitHintSpec = new SegmentsSplitHintSpec(null);
```

Comment:
Do you mean the constructor of `IngestSegmentFireshoseFactory`? The thing is, `maxInputSegmentBytesPerTask` was in the constructor (and is still there but deprecated), but the compaction task doesn't accept `IngestSegmentFireshoseFactory` or any firehose factory as its parameter. We need a way to configure `maxInputSegmentBytesPerTask` without specifying the firehose factory. 

Another way could be adding `maxInputSegmentBytesPerTask` as a parameter of the compaction task. Maybe this could be better?

Reply:
Another concern is, `maxInputSegmentBytesPerTask` sounds like a tuning config, but it's in the firehose constructor currently.